modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase

```diff
@@ -403,44 +403,100 @@ definition {
+				subject = "Subject Field");
 
 			PortletEntry.save();
 		}
 
-		task ("When the user edits the template") {
+		task ("When a custom object with a custom field are created") {
+			ObjectAdmin.addObjectViaAPI(
+				labelName = "Custom Object 161918",
+				objectName = "CustomObject161918",
+				pluralLabelName = "Custom Objects 161918");
+
+			ObjectAdmin.addObjectFieldViaAPI(
+				fieldBusinessType = "Text",
+				fieldLabelName = "Custom Field",
+				fieldName = "customObjectField",
+				fieldType = "String",
+				isRequired = "false",
+				objectName = "CustomObject161918");
+
+			ObjectAdmin.publishObjectViaAPI(objectName = "CustomObject161918");
+		}
+
+		task ("And when a action on after add to send a email are created") {
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectPortlet.selectCustomObject(label = "Custom Object 161918");
+
+			ObjectAdmin.goToActionsTab();
+
+			ObjectAdmin.addObjectActionViaUI(
+				actionLabel = "Custom Action",
+				active = "true",
+				inputMethod = "true",
+				inputValue = "Action shot on add",
+				notificationName = "Notification Template",
+				objectLabel = "Custom Object 161918",
+				thenAction = "Notification",
+				whenAction = "On After Add");
+		}
+
+		task ("And when the user edits the template") {
+			ApplicationsMenu.gotoPortlet(
+				category = "Notifications",
+				panel = "Control Panel",
+				portlet = "Templates");
+
 			ObjectNotifications.editNotificationTemplate(
-				emailBody = "Email Body Edit",
+				emailBody = "Email Body Content Edited",
 				entryTo = "test2@liferay.com",
-				externalReferenceCode = "ERC Edit",
-				fromAddress = "Liferay Edit",
-				fromName = "Test Test Edit",
-				name = "New Notification Edit",
+				externalReferenceCode = "ERC Edited",
+				fromAddress = "test2@liferay.com",
+				fromName = "Test Test Edited",
+				name = "Notification Template Edited",
 				notificationTemplateName = "Notification Template",
-				subject = "Subject Edit");
+				subject = "Subject Field Edited");
 
 			PortletEntry.save();
 		}
 
-		task ("Then the notification template will have all edited values") {
+		task ("And when an entry is added on the custom object") {
+			ObjectAdmin.addObjectSingleFieldEntryViaAPI(
+				fieldName = "customObjectField",
+				objectName = "CustomObject161918",
+				value = "Trigger");
+		}
+
+		task ("Then the notification template will have all edited values and the email message too") {
 			ObjectNotifications.viewNotificationTemplateDetails(
-				emailBody = "Email Body Edit",
+				emailBody = "Email Body Content Edited",
 				entryTo = "test2@liferay.com",
-				externalReferenceCode = "ERC Edit",
-				fromAddress = "Liferay Edit",
-				fromName = "Test Test Edit",
-				name = "New Notification Edit",
-				subject = "Subject Edit");
+				externalReferenceCode = "ERC Edited",
+				fromAddress = "test2@liferay.com",
+				fromName = "Test Test Edited",
+				name = "Notification Template Edited",
+				subject = "Subject Field Edited");
```

Comment:
Move this to after editing the email template

`Then the email notification template is updated`